### PR TITLE
Muon analysis updates 8.1.x

### PR DESCRIPTION
**Pull request by usarica**

### CODE REVIEW

**Comment by hroskes** [comment]
@vbotta you probably want to watch this (and I do too @hroskes)

**Comment by mmusich** [comment]
please test

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C [hunk 1]
```diff
@@ -88,7 +88,6 @@ void PlotAlignmentValidation::useFitForDMRplots(bool usefit)
 //------------------------------------------------------------------------------
 void PlotAlignmentValidation::setOutputDir( std::string dir )
 {
-  // we should check if this dir exsits...
   std::cout <<"'"<< outputDir <<"' = "<< dir << std::endl;
```
Comment (ghellwig) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @usarica Naive question: do you check if the directory exists?

File: Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py [hunk 2]
```diff
@@ -153,6 +153,8 @@ def getGeneral( self ):
         if not self.has_option(internal_section, "workdir"):
             self.set(internal_section, "workdir", "/tmp/$USER")
         general["workdir"] = self.get(internal_section, "workdir")
+        for dir in "workdir", "datadir", "logdir", "eosdir":
+            general[dir] = os.path.expandvars(general[dir])
         return general
```
Comment (ghellwig) [inline Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py]: @usarica Please avoid, using variable names that override built-in functions (`dir`).

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C (hunk 1 above)
Comment (usarica) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: @hroskes?

File: Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py (hunk 2 above)
Comment (usarica) [inline Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py]: Please tag @hroskes as the original author of this commit as well. Thanks.

**Comment by ghellwig** [comment]
@usarica In addition to my inline comments:
Please provide a description of this PR and some kind of documentation?
Has it been tested or used in validations of recent/current alignment campaigns?

**Comment by usarica** [comment]
@ghellwig The PR fixes the mu+ mu- ordering, most relevant for applying asymmetric eta cuts. It was actually used in the Run 2015 EOY validations but was never propagated due to previous Zmumu PRs pending at that time (one is still pending).

**Comment by ghellwig** [comment]
@usarica Could you then update the description of the PR?
Concerning the pending PR you are mentionining: I guess, this (#14511) PR contains also the changes of the other pending PR, right?

File: Alignment/OfflineValidation/macros/PlotAlignmentValidation.C (hunk 1 above)
Comment (hroskes) [inline Alignment/OfflineValidation/macros/PlotAlignmentValidation.C]: 2 lines down I do mkdir (and 2nd argument=true means -p)

File: Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py (hunk 2 above)
Comment (hroskes) [inline Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py]: I can fix it in the next iteration, I think at this point it's not worth holding.  It's right before the return, so there's no danger of trying to use dir().

Comment (ghellwig) [inline Alignment/OfflineValidation/python/TkAlAllInOneTool/betterConfigParser.py]: @hroskes right, in this case it does not matter.
And yes, please fix it in the next iteration.

**Comment by ghellwig** [comment]
+1
@usarica approving, but you need to add a description to this PR (at the top), such that one does not need to browse through the whole discussion

**Pull request merged**